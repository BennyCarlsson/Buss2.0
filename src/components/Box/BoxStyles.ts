import styled from "styled-components"

export const Box = styled.div`
  border: 2px solid #a3a3a3;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 20vh;
  border-radius: 20px;
`
