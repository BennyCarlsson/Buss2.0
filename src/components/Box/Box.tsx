import React from "react"
import { Box as StyledBox } from "./BoxStyles"

const Box: React.FC = (props) => {
  return <StyledBox>{props.children}</StyledBox>
}

export default Box
