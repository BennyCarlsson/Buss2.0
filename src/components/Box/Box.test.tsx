import React from "react"
import { shallow } from "enzyme"
import renderer from "react-test-renderer"
import { Box as StyledBox } from "./BoxStyles"
import Box from "./Box"

describe("Snapshot Tests", () => {
  it("render Box component", () => {
    const tree = renderer.create(<Box />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("render Box with children component", () => {
    const tree = renderer
      .create(
        <Box>
          <p>Hello World</p>
        </Box>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe("Box", () => {
  it("render Box", () => {
    const wrapper = shallow(<Box />)
    expect(wrapper.contains(<StyledBox></StyledBox>)).toEqual(true)
  })
  it("render Box", () => {
    const wrapper = shallow(<Box>
      13:37
    </Box>)
    expect(wrapper.text()).toEqual("13:37")
  })
})
