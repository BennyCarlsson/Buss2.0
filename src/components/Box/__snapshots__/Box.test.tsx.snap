// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests render Box component 1`] = `
<div
  className="sc-bdnylx boKVLG"
/>
`;

exports[`Snapshot Tests render Box with children component 1`] = `
<div
  className="sc-bdnylx boKVLG"
>
  <p>
    Hello World
  </p>
</div>
`;
